### PR TITLE
Clean up on game join/player respawn packets

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Review by Earthcomputer — APPROVED** [review]

File: mappings/net/minecraft/network/packet/s2c/play/GameJoinS2CPacket.mapping [hunk 1]
```diff
@@ -4,18 +4,33 @@ CLASS net/minecraft/class_2678 net/minecraft/network/packet/s2c/play/GameJoinS2C
 	FIELD field_12280 playerEntityId I
 	FIELD field_12281 hardcore Z
 	FIELD field_12282 gameMode Lnet/minecraft/class_1934;
-	FIELD field_12284 dimensionTracker Lnet/minecraft/class_5318$class_5319;
+	FIELD field_12284 registryTracker Lnet/minecraft/class_5318$class_5319;
+		COMMENT Accessor to dimension types.
```
Comment (Earthcomputer) [inline mappings/net/minecraft/network/packet/s2c/play/GameJoinS2CPacket.mapping:8]:
```suggestion
		COMMENT Accessor to synced registries.
```

File: mappings/net/minecraft/network/packet/s2c/play/GameJoinS2CPacket.mapping [hunk 2]
```diff
@@ -4,18 +4,33 @@ CLASS net/minecraft/class_2678 net/minecraft/network/packet/s2c/play/GameJoinS2C
...
 	FIELD field_20666 flatWorld Z
 	FIELD field_24618 reducedDebugInfo Z
 	FIELD field_24619 showDeathScreen Z
-	FIELD field_25134 dimensionId Lnet/minecraft/class_5321;
+	FIELD field_25134 world Lnet/minecraft/class_5321;
+		COMMENT The key of the world which the player joins.
+	FIELD field_25320 worlds Ljava/util/Set;
+		COMMENT A collection of possible worlds on the server, currently only used by
+		COMMENT clientside command suggestions before server sends an accurate list of
+		COMMENT suggestions.
```
Comment (Earthcomputer) [inline mappings/net/minecraft/network/packet/s2c/play/GameJoinS2CPacket.mapping:20]:
```suggestion
		COMMENT A collection of possible worlds on the server, currently only used by
		COMMENT clientside command suggestions.
```
It doesn't ask the server for a more accurate list of suggestions.

**Review by Earthcomputer — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/network/packet/s2c/play/GameJoinS2CPacket.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/network/packet/s2c/play/GameJoinS2CPacket.mapping:8]: For now this `RegistryTracker.Modifiable` is hardcoded (yes, final) to support dimension types only, so.

**Comment by liach** [comment]
Closing for https://github.com/FabricMC/yarn/commit/c8192e4d369f0ef977bd123833248ac1da384753#diff-6a22d519560ac2eb75d7c5b07339f3c6R7

**Pull request closed without merging**